sync3/handler/ensure_polling.go

```diff
@@ -58,19 +58,31 @@ func NewEnsurePoller(notifier pubsub.Notifier, enablePrometheus bool) *EnsurePol
 }
 
 // EnsurePolling blocks until the V2InitialSyncComplete response is received for this device. It is
-// the caller's responsibility to call OnInitialSyncComplete when new events arrive. Returns the Success field from the
-// V2InitialSyncComplete response, which is true iff there is an active poller.
+// the caller's responsibility to call OnInitialSyncComplete when new events arrive. Returns whether
+// or not the token is expired
 func (p *EnsurePoller) EnsurePolling(ctx context.Context, pid sync2.PollerID, tokenHash string) bool {
 	ctx, region := internal.StartSpan(ctx, "EnsurePolling")
 	defer region.End()
 	p.mu.Lock()
-	// do we need to wait?
-	if p.pendingPolls[pid].done {
+	// do we need to wait? Expired devices ALWAYS need a fresh poll
+	expired := p.pendingPolls[pid].expired
```

Comment:
If pid is missing from the map then expired is false. Is this correct? I think so?

Reply:
Oh it looks like this function isn't writing into the pendingPolls map at all(?)

Reply:
If pid is missing, then expired is false, and this is correct insofar that the if statement below should just care about if it is "done" or not, which will be false for missing pids. So yes, happy this is fine.